src/promptflow/promptflow/batch/_csharp_executor_proxy.py

```diff
@@ -0,0 +1,49 @@
+import subprocess
+from pathlib import Path
+from typing import Any, Mapping, Optional
+
+from promptflow.batch._base_executor_proxy import APIBasedExecutorProxy
+from promptflow.executor._result import AggregationResult
+from promptflow.storage._run_storage import AbstractRunStorage
+
+EXECUTOR_PORT = 8080
+
+
+class CSharpExecutorProxy(APIBasedExecutorProxy):
```

Comment:
can you name it `CSharpAPIBasedExecutorProxy`? if I understand it correctly, maybe I will add a `CSharpExecutorProxy` which directly call CSharp dll for local `flow test`

Reply:
And to confirm, you will directly merge this to main? This is C# only code to my understanding and I suppose you will make it in feature branch only for a time + @thy09

Reply:
Executor proxy is the multilingual contract in the batch engine. I prefer using `language+ExecutorPorxy` as the name, like the existing `PythonExecutorProxy`.

Reply:
`CSharpExecutorProxy` may not be merged into main. But `APIBasedExecutor` should be merged since the runtime will inherit it. 

But I think it can also be merged into main. Do you have any concern?